module/VuFind/src/VuFind/Db/Service/OaiResumptionService.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Database service for OaiResumption.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Service;
+
+use Laminas\Log\LoggerAwareInterface;
+use VuFind\Db\Entity\OaiResumptionEntityInterface;
+use VuFind\Db\Table\DbTableAwareInterface;
+use VuFind\Db\Table\DbTableAwareTrait;
+use VuFind\Log\LoggerAwareTrait;
+
+/**
+ * Database service for OaiResumption.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+class OaiResumptionService extends AbstractDbService implements
+    DbTableAwareInterface,
+    LoggerAwareInterface,
+    OaiResumptionServiceInterface
+{
+    use DbTableAwareTrait;
```

Comment:
As in other PRs, I'm using DbTableAwareTrait to avoid creating a factory here that will need to be removed in #2233. Otherwise I would prefer more explicit dependency injection.